include/os/freebsd/spl/sys/vnode.h

```diff
@@ -83,6 +85,22 @@ vn_is_readonly(vnode_t *vp)
 #define	vn_has_cached_data(vp)	\
 	((vp)->v_object != NULL && \
 	(vp)->v_object->resident_page_count > 0)
+
+static __inline void
+vn_flush_cached_data(vnode_t *vp, boolean_t sync)
+{
+#if __FreeBSD_version >= 1300125
+	if (vm_object_mightbedirty(vp->v_object)) {
+#else
+	if (vp->v_object && vp->v_object->flags & OBJ_MIGHTBEDIRTY) {
```

Comment:
We could assume `v_object` is valid here. It's checked by `vn_has_cached_data` at the call site, and `vm_object_mightbedirty` also assumes the object is valid.

Reply:
Updated.